Issues-#31 프로젝트 세팅

KJW921222 opened the pull request:
## 개요
애플리케이션의 개발을 위해 필요한 의존성을 추가하고, db 및 mybatis를 세팅할 필요가 있다.

## 작업사항
1. application.properties 삭제

2. application.yml 작성
- db 연결정보 작성
- mybatis 설정 (mapper-locations, map-underscore-to-camel-case)

3. build.gradle 의존성 추가
- validation
- mybatis
- lombok
- devtools
- mariadb

## 기타
security는 학습 후 다룰 내용이라 생각하여 주석 처리함.

sunggwon committed:
프로젝트 생성

sunggwon committed:
Update README.md

초안 작성

sunggwon committed:
Update README.md

api 목록 초안 작성

sunggwon committed:
Update README.md

API 목록 오타 수정

sunggwon committed:
Update README.md

API 목록 오타 수정

sunggwon committed:
Update README.md

API 링크 제대로 안 걸리는 문제 수정

sunggwon committed:
Update README.md

API 목록 링크 에러 처리

sunggwon committed:
Update README.md

1차 협의 내용

sunggwon committed:
Update README.md

데이터 베이스 모델링 추가

KJW921222 committed:
Update README.md

sunggwon committed:
Update README.md

깃 전략 내용 추가

sunggwon committed:
Update README.md

sunggwon committed:
Update README.md

프로젝트 범위 지정

sunggwon committed:
Revert "프로젝트 생성"

This reverts commit 921a2d929ecb975dc478be8dec87be25de1873fb.

sunggwon committed:
프로젝트 생성

small 프로젝트 생성
  - openJDK 1.8
  - Spring Boot 2.6.6

f-lab-robert committed:
Merge pull request #18 from f-lab-edu/feature-#13

Issues-#13 README 작성

KJW921222 committed:
Merge pull request #20 from f-lab-edu/feature-#16

Issues-#16 프로젝트 생성

KJW921222 committed:
프로젝트 세팅

########
1. application.properties 삭제
2. application.yml 작성
3. build.gradle 의존성 추가

########
issue#31

f-lab-robert commented on build.gradle:
```diff
@@ -0,0 +1,32 @@
+    implementation 'org.springframework.boot:spring-boot-starter-validation'
+    implementation 'org.springframework.boot:spring-boot-starter-web'
+    implementation 'org.mybatis.spring.boot:mybatis-spring-boot-starter:2.2.2'
+    //implementation 'org.springframework.boot:spring-boot-starter-thymeleaf'
```
이 부분은 아예 없어도 될 것 같습니다. 차후 PR 에서 제거해주세요.

sunggwon replied:
thymeleaf 주석 제거하였습니다.

f-lab-robert commented on src/main/resources/application.yml:
```diff
@@ -0,0 +1,12 @@
+    url: jdbc:mariadb://34.125.26.207:3306/small
+    driver-class-name: org.mariadb.jdbc.Driver
+    username: root
+    password: small#1234
```
- root 유저 사용은 지양해야합니다. root 는 다른 유저들을 통제하기 위한 막강한 유저 입니다. 다른 user 를 사용해주세요.
- password 가 노출되면 안되겠죠 ? .gitignore 를 통해 application.yml 이 올라오지 않도록 해주세요. 더불어, 현재 pwd 는 노출되었으니 변경해서 다시 올려주세요.
- 더불어, 저와 같은 제 3자는 application.yml 이 아예 없으면 프로젝트 셋업에 어려움이 있을 수 있으니, applicaton-xxx.yml 등 다른 방법을 통해 application.yml 의 형태만 볼 수 있는 example 파일이 하나 올라오면 좋을 것 같다는 생각이 드네요. 가령, 다음과 같은 example 형태의 application-xxx.yml 은 올려주시되, 실제 value 를 포함하고 있는 application.yml 은 gitignore 를 통해 제거해주시는게 좋겠네요, 
```
application-xxx.yml

spring:
  datasource:
    url: jdbc:mariadb://34.125.26.207:3306/small
    driver-class-name: org.mariadb.jdbc.Driver
    username: xxx
    password: xxxx
```

f-lab-robert replied:
fyi - 주원님이 패스워드 및 유저네임을 암호화 없이 "문자열" 로 사용하는 것에 대해 안전한지 질문을 주셨는데. 관련해서는 멘토링 시간에 조금 더 설명을 드리도록 할게요. 우선은 위의 가이드를 따라주세요.

sunggwon replied:
.gitIgnore에 추가한 후 application-template.yml 파일을 추가하였습니다.
또한 기존에 노출된 root 패스워드도 변경 완료했습니다.

f-lab-robert commented on src/main/resources/application.yml:
```diff
@@ -0,0 +1,12 @@
+  #type-aliases-package:
+  mapper-locations: classpath:mapper/*.xml
+  configuration:
+    map-underscore-to-camel-case: true
```
no new line 이라는 말이 있는데요. 어떤걸 의미하는걸까요 ?

sunggwon replied:
IEEE의 POSIX(UNIX OS의 공통 API)에서 문서 가장 끝에 개행문자가 나오는 것을 줄(line)으로 정의하였고
예전 gcc 컴파일러는 POSIX에서 정의 된 줄 단위로 코드를 읽었기 때문에 에러가 발생할 수 있다. 
요즘 컴파일러는 해당 문제가 없지만 넣어주는 것이 좋다.
따라서 저희 프로젝트에도 .editorconfig의 insert_final_newline 를 통해 IDE에 자동으로 개행을 넣어주는 것을 기능을 추가하였습니다.

f-lab-robert commented on build.gradle:
```diff
@@ -0,0 +1,32 @@
+
+group = 'com.flabedu.small'
+version = '0.0.1-SNAPSHOT'
+sourceCompatibility = '17'
```
Java 17 을 선택한 이유에 대해서도 기재가 되면 좋을 것 같네요.

sunggwon replied:
코드에 주석으로 java 17을 선택한 이유 추가해놨습니다.

f-lab-robert replied:
@sunggwon 학습을 위해 최근 자바 기능을 사용하기 위해 버전 17로 지정

위와 같은 코멘트는 좋지 않습니다. 선택지가 무엇이 있었으며 그 중에 17의 경우 어떤게 달라졌기 때문에 그 부분에 대해 경험해보고 싶다는 구체적인 이유가 있는게 좋습니다. 재차 점검해보시겠어요 ?

f-lab-robert replied:
@sunggwon 

> 선택지가 무엇이 있었으며 
```
EX )
 Java8, 11, 17.... 등등 LTS 이며... LTS 는 무엇이며. LTS 를 사용하지 않았을 경우에 ... 문제가 있을 수 있으며 등등 
```

> 17의 경우 어떤게 달라졌기 때문에

```
EX ) 
GC, JVM ...... 등등 java8 은 포기, 단 Java 8 에서 많은 ...... 등의 기능이 새로 추가됨
11 의 경우 GC,JVM 은 .... 이는 17과 동일 혹은 다름. 때문에 11,17 을 고민함.
17은 조금 더 문법적 추가 기능이 있으나 여러 프레임워크 등과의 호환성 문제 등이 있을 수 있음 등.... 
단, 현 프로젝트는 Spring 으로 진행하며 학습의 목적이 크기에 ... 등등 
```



- 재차 위의 요구 사항을 다시 읽어보시겠어요 ? 단순 진열하신 4개의 기능을 사용하기 위해 선택했다는 걸로 밖에 보이지 않습니다. 각각 java 버전에서 사용하는 gc 및 jvm 의 구조는 무엇이며 등 저희가 배웠던 지식들이 중요 기준이 되어야 하지 않을까요 ?
적어주신 기준에 의하면 프로젝트를 진행하다가 아래 4개의 기능을 사용하지 않게되면 차후에 11로 버전을 돌리실 예정이신가요 ? 
- java 의 신에서 학습하신 내용들을 잘 떠올려보시면 제가 항상 요구하는 내용들은 단순 기능들이 아닌 모든 선택지의 중요한 특징들을 정리하는 것으로 부터 시작합니다. java8,11,17 의 gc,jvm,기능의 중요 변곡점들에 대해 정리를 하는게 선행되어야 할 것 같습니다.
- 앞으로 리뷰 요청 전에 이러한 부분들을 한번 더 생각해보시면 좋을 것 같습니다.

KJW921222 replied:
자바 17을 선택한 이유 수정했습니다.

sunggwon committed:
- application.yml 파일 gitIgnore에 적용
- 셋업을 위한 템플릿으로 application_template.yml 추가
- build.gradle의 thumeleaf 종속성 주석 제거
- no new line 방지를 위한 .editorconfig 추가

sunggwon committed:
.gitignore 적용

sunggwon committed:
.gitIgnore 적용
 - 기존 push 된 프로젝트의 gitIgnore 적용을 위해 재 커밋

f-lab-robert commented on src/main/resources/application_template.yml:
```diff
@@ -0,0 +1,12 @@
+spring:
+  datasource:
+    url: jdbc:mariadb://34.125.26.207:3306/small
```
@sunggwon 이 부분도 노출이 되면 안될 것 같네요.

sunggwon committed:
build.gradle
  - java 버전 선정 이유 구체적으로 기술
application_template.yml
  - 보안상 문제가 발생할 수 있는 부분 처리

KJW921222 committed:
프로젝트 세팅

########
1. build.gradle에 자바 17을 채택한 이유 작성

########
issue#31

KJW921222 committed:
프로젝트 세팅

########
1. build.gradle에 자바 17을 채택한 이유 작성

########
issue#31
